packages/mui-utils/src/useForkRef.ts

```diff
@@ -1,10 +1,10 @@
 import * as React from 'react';
 import setRef from './setRef';
 
-export default function useForkRef<Instance>(
-  refA: React.Ref<Instance> | null | undefined,
-  refB: React.Ref<Instance> | null | undefined,
-): React.Ref<Instance> | null {
+export default function useForkRef<InstanceA, InstanceB>(
+  refA: React.Ref<InstanceA> | null | undefined,
+  refB: React.Ref<InstanceB> | null | undefined,
+): React.Ref<InstanceA & InstanceB> | null {
```

Comment:
This helps TypeScript with inference with the proposed changes in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/58936.

Reply:
Nice, this will basically create a union of the two in case they are different 👍